### ISSUE TITLE
Bug: ESLint via npm exec requires elevation, but shouldn't (Win11 21H2)

### ISSUE THREAD

**Issue by Keyacom** [eslint/eslint#16338]
### Environment

Node version: v18.9.0
npm version: v8.19.2
Local ESLint version: v8.23.1 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22000

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
// None yet, just assumed defaults.
```
</details>

```js
const consoleWindow = document.querySelector("#con");

const {
  ioInput, ioOutput, ioIbefore, ioObefore
} = consoleWindow.dataset;
```

### What did you expect to happen?

```
C:\Users\Keyacom\Documents\GitHub\pages-from-school\psi\grade2\reklama-anim\script.js
  1:1  error  Parsing error: The keyword 'const' is reserved

✖ 1 problem (1 error, 0 warnings)
```
This was the actual problem with the file. This was before I added any rules to `.eslintrc`.

### What actually happened?

When run without elevation, this is the output (even though Node's dir is in my PATH):
```
'"node"' is not recognized as an internal or external command,
operable program or batch file.
```
It ran as expected when in an elevated shell.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Keyacom** [comment]
The title does mention `npm exec`, but this is in `npx` as well. The reason I mentioned the former one is because the npm/npx repo is archived, since npx is now part of npm/cli.

**Comment by nzakas** [comment]
This doesn't seem like it's ESLint-specific. Are you able to run other Node.js-powered command line utilities like Prettier?

**Comment by Keyacom** [comment]
> This doesn't seem like it's ESLint-specific. Are you able to run other Node.js-powered command line utilities like Prettier?

I will check this tomorrow after school.

**Comment by Keyacom** [comment]
As it turns out, this is not explicitly a problem with ESLint, but with npm exec.

**Comment by nzakas** [comment]
Great, thanks for confirming. Closing as this is not related to ESLint directly.
